File: backend/src/main/scala/services/review/objects/ReviewStatus.scala
package services.review.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}

enum ReviewStatus:
  case Active, Revoked

object ReviewStatus:
  private val enumLabel = text("ReviewStatus")
  def render(value: ReviewStatus): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[ReviewStatus] = parseEnumValue(value, ReviewStatus.values)
  given Encoder[ReviewStatus] = enumEncoder
  given Decoder[ReviewStatus] = enumDecoder(ReviewStatus.values, enumLabel)

File: frontend/src/objects/review/ReviewStatus.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
export const REVIEW_STATUS = {
  active: 'Active',
  revoked: 'Revoked',
} as const

export type ReviewStatus = (typeof REVIEW_STATUS)[keyof typeof REVIEW_STATUS]
